pkg/process/metadata/parser/java/util.go

```diff
@@ -0,0 +1,146 @@
+func (y *mapSource) GetDefault(key string, defVal string) string {
+	val, ok := y.m[key]
+	if !ok {
+		return defVal
+	}
+	return val
+}
+
+// newArgumentSource a PropertyGetter that is taking key=value from the list of arguments provided
+// it can be done to parse both java system properties (the prefix is `-D`) or spring boot property args (the prefix is `--`)
+func newArgumentSource(arguments []string, prefix string) *mapSource {
+	parsed := make(map[string]string)
+	for _, val := range arguments {
+		if !strings.HasPrefix(val, prefix) {
+			continue
+		}
+		parts := strings.SplitN(val[len(prefix):], "=", 2)
+		if len(parts) == 1 {
+			parsed[parts[0]] = ""
+		} else {
+			parsed[parts[0]] = parts[1]
+		}
+	}
+	return &mapSource{parsed}
+}
+
+// newPropertySourceFromStream create a PropertyGetter by selecting the most appropriate parser giving the file extension.
+// An error will be returned if the filesize is greater than maxParseFileSize
+func newPropertySourceFromStream(rc io.Reader, filename string, filesize uint64) (*props.PropertyGetter, error) {
+	if filesize > maxParseFileSize {
+		return nil, fmt.Errorf("unable to parse %q. max file size exceeded(actual: %d, max: %d)", filename, filesize, maxParseFileSize)
+	}
+	var properties props.PropertyGetter
+	var err error
+	ext := strings.ToLower(filepath.Ext(filename))
+	switch ext {
+	case ".properties":
+		properties, err = props.Read(rc)
+	case ".yaml", ".yml":
+		properties, err = newYamlSource(rc)
+	default:
+		return nil, fmt.Errorf("unhandled file type for %q", filename)
+	}
+	return &properties, err
+}
+
+// newPropertySourceFromFile wraps filename opening and closing, delegating the rest of the logic to newPropertySourceFromStream
+func newPropertySourceFromFile(filename string) (*props.PropertyGetter, error) {
+	f, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	fi, err := f.Stat()
+	if err != nil {
+		return nil, err
+	}
+	return newPropertySourceFromStream(f, filename, uint64(fi.Size()))
+}
+
+// longestPathPrefix extracts the longest path's portion that's not a pattern (i.e. /test/**/*.xml will return /test/)
+func longestPathPrefix(pattern string) string {
+	idx := strings.IndexAny(pattern, "?*")
+	if idx < 0 {
+		return pattern
+	}
+	idx = strings.LastIndex(pattern[:idx], "/")
+	if idx < 0 {
+		return ""
+	}
+	return pattern[:idx]
+}
+
+// scanSourcesFromFileSystem returns all the PropertyGetter sources built from files matching profilePatterns.
+// profilePatterns is a map that has for key the name of the spring profile and for key the values of patterns to be evaluated to find those files
+func scanSourcesFromFileSystem(profilePatterns map[string][]string) map[string]*props.Combined {
+	ret := make(map[string]*props.Combined)
+	matcher := antpath.New()
+	for profile, pp := range profilePatterns {
+		for _, pattern := range pp {
+			startPath := longestPathPrefix(pattern)
```

Comment:
Would using `filepath.Glob` be simpler here? i.e., would you need `longestPathPrefix` anymore?